dolosse/generic_json_consumer.py

```diff
@@ -0,0 +1,142 @@
+        print("Using default topic name 'default'...")
+    else:
+        topic = sys.argv[1]
+    return topic
+
+
+def redraw(figures, key):
+    """Redraw the given figure from a dict of figures"""
+    figures[key].canvas.draw()
+    figures[key].canvas.flush_events()
+
+
+
+
+
+def main():
+    """Main function - reads json data from a kafka topic and graphs it"""
+
+    screen = curses.initscr()
+    curses.cbreak()
+    screen.nodelay(True)
+
+    topic = check()
+
+    data = {}
+    line = {}
+    plots = {}
+    figures = {}
+    axes = {}
+    x_axis_data = []
+    count = 0
+    pyplot.ion() #Interactive mode on
+    # To consume latest messages and auto-commit offsets
+    consumer = kafka.KafkaConsumer(topic,
+                                   group_id='my-group',
+                                   bootstrap_servers=['localhost:9092'],
+                                   auto_offset_reset='earliest',
+                                   enable_auto_commit=False)
+
+    exit_char = curses.ERR
+
+    for message in consumer:
+        try:
+            update = json.loads(message.value)
+        except json.decoder.JSONDecodeError as problem:
+            print("Error: Flawed JSON in value read from kafka topic: ",
+                  message.value)
+            print("Error message: ", problem)
+            update = ""
+        data.update(update)
+        print("%s:%d:%d: key=%s value=%s" % (message.topic, message.partition,
+                                             message.offset, message.key,
+                                             data))
+        x_axis_data.append(count)
+        count += 1
+
+        populate(line, data, x_axis_data)
+
+        for key in line:
+            while len(line[key]) < len(x_axis_data):
+                line[key].append(0) #Variable went missing from data.
+                #Buffer with zeroes for now.
+            print(key, line[key])
+            if key in plots:
+                axes[key][0].set_ydata(line[key])
+                axes[key][0].set_xdata(x_axis_data)
+                plots[key].relim()
+                plots[key].autoscale_view()
+                redraw(figures, key)
+            else:
+                figures[key], plots[key] = pyplot.subplots()
+                axes[key] = plots[key].plot(line[key])
+                plots[key].set_autoscaley_on(True)
+                plots[key].set_autoscalex_on(True)
+                plots[key].set_title(key)
+        while (consumer.end_offsets([kafka.TopicPartition(topic, 0)])
+               [kafka.TopicPartition(topic, 0)]
+               == consumer.position(kafka.TopicPartition(topic, 0))):
+            for key in line:
+                redraw(figures, key)
+            time.sleep(0.2)
+            exit_char = screen.getch()
+            if exit_char != curses.ERR:
+                break
+        if exit_char != curses.ERR:
+            break
+        exit_char = screen.getch()
+
+
```

Comment:
Should have at most two empty lines here.

Reply:
Since that's not between functions, I cut it down to one empty line.